processing/src/main/java/org/apache/druid/segment/filter/OrFilter.java

```diff
@@ -71,6 +75,167 @@ public OrFilter(List<Filter> filters)
     this(new LinkedHashSet<>(filters));
   }
 
+  @Override
+  public <T> FilterBundle makeFilterBundle(
+      ColumnIndexSelector columnIndexSelector,
+      BitmapResultFactory<T> bitmapResultFactory,
+      int selectionRowCount,
+      int totalRowCount,
+      boolean includeUnknown,
+      boolean allowPartialIndex
+  )
+  {
+    final List<T> indexes = new ArrayList<>();
+    final List<Filter> indexFilters = new ArrayList<>();
+    final List<Filter> matcherFilters = new ArrayList<>();
+
+    final long bitmapConstructionStartNs = System.nanoTime();
+    for (Filter subfilter : filters) {
+      boolean needMatcher = true;
+      final BitmapColumnIndex index = subfilter.getBitmapColumnIndex(columnIndexSelector);
+      if (index != null) {
+        final T bitmapResult = index.computeBitmapResult(
+            bitmapResultFactory,
+            selectionRowCount,
+            totalRowCount,
+            includeUnknown
+        );
+        if (bitmapResult != null) {
+          if (!bitmapResultFactory.isEmpty(bitmapResult)) {
+            indexes.add(bitmapResult);
+          }
+          indexFilters.add(subfilter);
+          needMatcher = !index.getIndexCapabilities().isExact();
```

Comment:
Should it be something like:
```suggestion
          needMatcher |= !index.getIndexCapabilities().isExact();
```